app/routes.py

```diff
@@ -5,52 +5,60 @@
 import scripts.clustering.clustering as clustering
 from app.data.clustering import get_smiles_json
 from app.data.color_functions import color_hex_to_array
+import pandas as pd
 
 @app.route('/')
-@app.route('/index')
-def index():
+@app.route('/index', methods=['GET', 'POST'])
+def index():     
+  session.clear()
+  if request.method == 'POST':
+    try:
+      data = request.get_array(field_name='file')
+      smiles, include_mpo = construct_smiles(data)
+      session['smiles'] = smiles
+      session['include_mpo'] = include_mpo
+
+      if include_mpo:
+        print(smiles)
+        unique_compounds = pd.DataFrame(dict((k, [v.get('mpo', ''), v['label']]) for k, v in smiles.items()), index=['mpo', 'label']).T
```

Comment:
I updated the dictionary names in my last PR, so this should be how to get the MPO score (sorry about that!)
```suggestion
        unique_compounds = pd.DataFrame(dict((k, [v.get('property', ''), v['label']]) for k, v in smiles.items()), index=['mpo', 'label']).T
```